### PR TITLE
Migrate example to functional components

### PR DESCRIPTION
**Pull request by marchenk0va**
Issue [#1808](https://github.com/callstack/react-native-paper/issues/1808)

Migrated example app to functional components, used reducer hook for managing state.

### CODE REVIEW

File: example/src/Examples/DialogExample.tsx [hunk 1]
```diff
@@ -9,99 +9,56 @@ import {
...
-    visible2: false,
-    visible3: false,
-    visible4: false,
-    visible5: false,
-  };
+  const _closeDialog1 = () => setVisible1(true);
+  const _closeDialog2 = () => setVisible2(true);
+  const _closeDialog3 = () => setVisible3(true);
+  const _closeDialog4 = () => setVisible4(true);
+  const _closeDialog5 = () => setVisible5(true);
```
[the remark this answers is not in the document]
Comment (jbinda) [reply example/src/Examples/DialogExample.tsx:29]: Also how about changing to below ?

<details>

```jsx

type ButtonVisibility= {
  [key: string]: boolean | undefined;
}

const DialogExample = () => {
  const [visible, setVisible] = React.useState<ButtonVisibility>({});

  const _toggleDialog = (name: string) => () => setVisible({ ...visible, [name]: !visible[name]});

  const _getVisible = (name: string) => !! visible[name];

  const {
    colors: { background },
  } = useTheme();

  return (
    <View style={[styles.container, { backgroundColor: background }]}>
      <Button mode="outlined" onPress={_toggleDialog('dialog1')} style={styles.button}>
        Long text
      </Button>
      <Button mode="outlined" onPress={_toggleDialog('dialog2')} style={styles.button}>
        Radio buttons
      </Button>
      <Button mode="outlined" onPress={_toggleDialog('dialog3')} style={styles.button}>
        Progress indicator
      </Button>
      <Button mode="outlined" onPress={_toggleDialog('dialog4')} style={styles.button}>
        Undismissable Dialog
      </Button>
      <Button mode="outlined" onPress={_toggleDialog('dialog5')} style={styles.button}>
        Custom colors
      </Button>
      <DialogWithLongText visible={_getVisible('dialog1')} close={_toggleDialog('dialog1')} />
      <DialogWithRadioBtns visible={_getVisible('dialog2')} close={_toggleDialog('dialog2')} />
      <DialogWithLoadingIndicator visible={_getVisible('dialog3')} close={_toggleDialog('dialog3')} />
      <UndismissableDialog visible={_getVisible('dialog4')} close={_toggleDialog('dialog4')} />
      <DialogWithCustomColors visible={_getVisible('dialog5')} close={_toggleDialog('dialog5')} />
    </View>
  );
};

```
</details>

File: example/src/Examples/MenuExample.tsx [hunk 2]
```diff
@@ -11,155 +11,133 @@ import {
...
+  >({ x: 0, y: 0 });
+
+  const _openMenu1 = () => setVisible1(true);
+  const _openMenu2 = () => setVisible2(true);
+  const _openMenu3 = () => setVisible3(true);
 
-  static title = 'Menu';
+  const _closeMenu1 = () => setVisible1(false);
+  const _closeMenu2 = () => setVisible2(false);
+  const _closeMenu3 = () => setVisible3(false);
```
Comment (jbinda) [inline example/src/Examples/MenuExample.tsx:41]: how about the same as in Dialog here ?

**Comment by jbinda** [comment]
Please check my test result. Tested also with Dark and RTL mode. I Have also drop some comments.

#### iOS


 - [x] ActivityIndicatorExample 
 - [x] AppbarExample 
 - [x] AvatarExample 
 - [x] BadgeExample 
 - [x] BannerExample 
 - [x] BottomNavigationExample 
 - [x] ButtonExample 
 - [x] CardExample 
 - [x] CheckboxExample 
 - [x] ChipExample 
 - [x] DataTableExample 
 - [ ] DialogExample 
      ❌ - seems not to hide after press OK. In `_closeDialog` handlers the flag passed is set to `true` and it should be `false`
 - [x] DividerExample 
 - [x] FABExample 
 - [x] IconButtonExample 
 - [x] ListAccordionExample
 - [x] ListAccordionExampleGroup
 - [x] ListSectionExample 
 - [x] MenuExample 
 - [x] ProgressBarExample 
 - [x] RadioButtonExample 
 - [x] RadioButtonGroupExample
 - [x] SearchbarExample 
 - [x] SnackbarExample 
 - [x] SurfaceExample
 - [x] SwitchExample 
 - [x] TextExample
 - [x] TextInputExample
 - [x] ToggleButtonExample
 - [x] TouchableRippleExample

#### Android


 - [x] ActivityIndicatorExample 
 - [x] AppbarExample 
 - [x] AvatarExample 
 - [x] BadgeExample 
 - [x] BannerExample 
 - [x] BottomNavigationExample 
 - [x] ButtonExample 
 - [x] CardExample 
 - [x] CheckboxExample 
 - [x] ChipExample 
 - [x] DataTableExample 
 - [ ] DialogExample 
      ❌ - seems not to hide after press OK. In `_closeDialog` handlers the flag passed is set to `true` and it should be `false`
 - [x] DividerExample 
 - [x] FABExample 
 - [x] IconButtonExample 
 - [x] ListAccordionExample
 - [x] ListAccordionExampleGroup
 - [x] ListSectionExample 
 - [x] MenuExample 
 - [x] ProgressBarExample 
 - [x] RadioButtonExample 
 - [x] RadioButtonGroupExample
 - [x] SearchbarExample 
 - [x] SnackbarExample 
 - [x] SurfaceExample
 - [x] SwitchExample 
 - [x] TextExample
 - [x] TextInputExample
 - [x] ToggleButtonExample
 - [x] TouchableRippleExample

#### Web

 - [x] ActivityIndicatorExample 
 - [x] AppbarExample 
 - [x] AvatarExample 
 - [x] BadgeExample 
 - [x] BannerExample 
 - [x] BottomNavigationExample 
 - [x] ButtonExample 
 - [x] CardExample 
 - [x] CheckboxExample 
 - [x] ChipExample 
 - [x] DataTableExample 
 - [ ] DialogExample 
      ❌ - seems not to hide after press OK. In `_closeDialog` handlers the flag passed is set to `true` and it should be `false`
 - [x] DividerExample 
 - [x] FABExample 
 - [x] IconButtonExample 
 - [x] ListAccordionExample
 - [x] ListAccordionExampleGroup
 - [x] ListSectionExample 
 - [x] MenuExample 
 - [x] ProgressBarExample 
 - [x] RadioButtonExample 
 - [x] RadioButtonGroupExample
 - [x] SearchbarExample 
 - [x] SnackbarExample 
 - [x] SurfaceExample
 - [x] SwitchExample 
 - [x] TextExample
 - [x] TextInputExample
 - [x] ToggleButtonExample
 - [x] TouchableRippleExample

**Comment by jbinda** [comment]
Apart of above I didn't found anything else ! Great job @marchenk0va ! 🎉

File: example/src/Examples/DialogExample.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (marchenk0va) [reply example/src/Examples/DialogExample.tsx:29]: Love this solution, thank you!

**Pull request merged**